Build with depend

janiversen opened the pull request:
<!-- Lines like this one are comments and will not be shown in the final output. -->
<!-- Make sure that you have read the "Contributing" section of the README and also the notes in CodingStyle. -->
<!-- If you are a collaborator, please add labels and assign other collaborators for a review. -->

### Describe the pull request:
<!-- Replace [ ] with [x] to select options. -->
- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
<!-- Describe your pull request in detail. -->
Removed local download/clone from scripts/build.sh and packaging/ios/build.sh
and replaced with get-dep-lib

Android is not touched, because I currently do not have a android test environment
(soon to be changed).

Feel free to make changes to my commits.

### Changes made:
<!-- Enumerate the changes with 1), 2), 3) etc. -->
<!-- Ensure the test cases are updated if needed. -->

### Related issues:
<!-- Reference issues with #<issue-num>. -->
<!-- Write "Fixes #<issue-num" to notify Github that this PR fixes an issue. -->

### Additional information:
<!-- Include sample dive log or other relevant information to allow testing the change where feasible. -->

### Release note:
<!-- Describe if this change needs a release note present in CHANGELOG.md. -->
<!-- Also, please make sure to add the release note on top of the file CHANGELOG.md. -->

### Documentation change:
<!-- If this PR makes changes to user functionality, then the documentation has to be updated too. -->
<!-- Please, briefly outline here what has changed in terms of the user experience (UX). -->
<!-- If UX changes have been made, a maintainer should apply the 'needs-documentation-change' label. -->

### Mentions:
<!-- Mention users that you want to review your pull request with @<user-name>. Leave empty if not sure. -->

dirkhh commented:
It looks like something is off with libgit2 - Mac, Linux, and Linux/qt55 are failing because of it.
I haven't looked into "why" at all. Happy to help, but I'm assuming that you have started to look into it already?

dirkhh commented:
Yes, Travis isn't perfect, but it sure helps a lot.
I'll simply wait for you here - this isn't urgent.

janiversen commented:
Do not ask me how, but when I look at the raw log of the failing jobs, I cannot find the call to get-dep-lib (even though e.g. my pushd commands are there). Can you help me explain that ?

janiversen commented:
I think I got it, can it be travis runs without -build-deps ? 

That would explain why I do not see the get-dep-lib call.

dirkhh commented:
Travis is controlled by https://github.com/Subsurface-divelog/subsurface/blob/master/.travis.yml
which calls three scripts under `scripts/${SUBSURFACE_PLATFORM}/ ` depending on which of the test environments it is. Those scripts then trigger the setup, the actual build, and what to do after the build succeeds.
Right now, for Mac, we switched to not building our dependencies and instead using them from Homebrew (because it ended up that that was much faster). The call to `build.sh` is here https://github.com/Subsurface-divelog/subsurface/blob/master/scripts/mac/travisbuild.sh#L18

dirkhh commented:
Oh, and the reason this is faster is that we have a setup that allows us to run Homebrew to build everything and then cache that - and the download of the cache is much faster than the build.
Obviously, the build needs to work in either of these two scenarios.
Equally obviously (I hope), for your testing you could simply modify Line 18 referenced in the previous comment and have it build things as you want to see them built.

janiversen commented:
Finally ! will integrate in the android build once this is done.

dirkhh commented on scripts/build.sh:
```diff
@@ -193,63 +193,70 @@ if [[ $PLATFORM = Darwin || "$LIBGIT" < "24" ]] ; then
 	# feature). So we painfully need to build the dependencies ourselves.
 
 	if [ "$BUILD_DEPS" == "1" ] ; then
```
so all of the changes in this commit are inside the `build-deps` condition. The commit message sounds like you are making changes for Travis, but of course this code isn't run on Travis.
I guess I'm confused...

janiversen commented:
I will implement your suggestions, and update the commit messages.

Right now I ran into a "naughty" problem. curl is accepted on all platforms except Android, do I really have to make a "if android use wget else use curl" ?

dirkhh commented:
We should be able to just install curl inside the container (note, Travis builds the Android apk inside a Ubuntu container, not inside the "outside VM" - so you need to install it as part of the `android-wrapper-script`, IIRC.

janiversen commented:
Cleaned up commits (the 2 commits earlier of scripts/build.sh were 2 different ideas) and texts.

Also implemented your comments.

Let us see, how much I have broken :-)

Will do the android part, once the general parts works.

dirkhh commented:
Android fails when building googlemaps plugin with 
```
/workspace/subsurface/packaging/android/build.sh: line 165: QMAKE: unbound variable
```

janiversen commented:
Already corrected, it was my semi-manual merge that took out too many lines (me blames my dm, because I am in parallel analysing my dives from earlier today)

janiversen commented:
This is not sufficient, how about I just quit the android integration. The rest should be ready to merge.

I can then try to setup a android build environment and test locally (but that will take some days)

wdyt?

dirkhh commented:
Works for me

dirkhh commented:
The nice thing about the way Android works on Travis is that it's really easy to recreate the exact same environment locally - simply install Ubuntu in a VM, install the packages listed and then run the scripts...

janiversen commented:
Yeah travis is GREEEN :-)

I am downloading ubuntu right now, but I suspect it will take a little while to get things working. I already have a android vm for testing. So the good thing (for the project) is that I should be able to e.g. try and bump the library versions myself.
